concent_api/verifier/tasks.py

```diff
@@ -293,6 +300,18 @@ def blender_verification_order(
         )
         return
 
+    if not check_if_sizes_in_pixels_are_equal(image_1, image_2):
+        log_string_message(
```

Comment:
I don't like the big blob of code here. I think I would prefer having some validation function that raises exception.

Reply:
If validation 'fails' in this function, the `verification_result` is returning with `VerificationResult.MISMATCH`. Look at the whole block (name already changed ):
```
       if not are_image_sizes_equal(image_1, image_2):
        log_string_message(
            logger,
            f'Blender verification failed. Sizes in pixels of images are not equal. SUBTASK_ID: {subtask_id}.'
            f'VerificationResult: {VerificationResult.MISMATCH.name}'
        )
        verification_result.delay(
            subtask_id,
            VerificationResult.MISMATCH.name,
        )
        return
```
I agree and don't like it as well, but I think setting of `verification_result` should be here. Logging could be moved to `are_image_sizes_equal` and changed to e.g. `are_image_sizes_equal_and_log_if_mismatch`... What do you think?

Reply:
I agree that it is written in current style of this task, so I would prefer to keep it as it is, and make flow changes like in refactor task. It might be added there as a comment.

Reply:
> Logging could be moved to are_image_sizes_equal and changed to e.g. are_image_sizes_equal_and_log_if_mismatch... What do you think?

No, definitely not. This looks like a generic utility funtion. It does not and should not know what its result is used for. I.e. that different sizes mean a failure of verification. That's the responsibility of the caller.

Reply:
BTW: `Sizes in pixels of images are not equal` would sound less awkward phrased as `Image resolutions differ`.

Reply:
I agree, but `resolution` is something different as `size in pixels`. Resolution says, how many `pixels` are in one `inch`. We could have two different images with sizes e.g. 40x50 cm and 30x60 cm, both in resolution e.g. `300 dpi` (dpi - pixels in inch), but sizes in pixels are different.
On the other hand, there can be two images - 30 x 30 cm in 300 dpi and 60 x 60 cm in 150 dpi and they are actually equal.
This function compares tuples with number of pixels. Now it will compare the whole `ndarray.shape` including number of color channels. 
I renamed the name to `are_image_sizes_and_color_channels_equal`, maybe the name is little bit awkward, but it is also very precise. If you want to rename it please let mi know.

Reply:
> I renamed the name to `are_image_sizes_and_color_channels_equal`, maybe the name is little bit awkward

It's fine. 

> I agree, but `resolution` is something different as `size in pixels`. Resolution says, how many `pixels` are in one `inch`. We could have two different images with sizes e.g. 40x50 cm and 30x60 cm, both in resolution e.g. `300 dpi` (dpi - pixels in inch), but sizes in pixels are different.
On the other hand, there can be two images - 30 x 30 cm in 300 dpi and 60 x 60 cm in 150 dpi and they are actually equal.

Yeah, you're right. Referring to image size as resolution is technically incorrect and pretty informal. "size in pixels" is more precise. But still the message is a bit awkward as is. Maybe `Image sizes in pixels differ.`?